### PR TITLE
Integrate patternlab into gulp tasks

### PR DESCRIPTION
**Pull request by 0vidiu**
@zenopopovici 
### WIP
This should not be considered for merge until the integration with patter-lab is complete.
At this moment there are some issues with the pattern-lab release which gives an JS error on opening.

**~~Errors in the console~~**
```
GET http://127.0.0.1:8080/styleguide/bower_components/classList.min.js net::ERR_ABORTED
(index):377
GET http://127.0.0.1:8080/styleguide/bower_components/classList.min.js 404 (Not Found)
```
**~~Error on the Front-End~~**
`[object Promise][object Promise][object Promise]`

### Steps to reproduce
0. (optional) `sudo npm install -g http-server`
1. `yarn`
2. `gulp`
3. `http-server ./www/patternlab` or open with other HTTP servers

### CODE REVIEW

**Comment by 0vidiu** [comment]
@zenopopovici 
### Patternlab implementation status:
* Got rid of the errors by downgrading patternlab and plugins versions (latest version is an alpha)
* Implemented assets copying from GulpFlow build to patternlab public folder (w.i.p)
* Implemented clean method
* Added two test atoms to verify that it works with data json files
* Improved the watchers

Still **w.i.p**

**Comment by 0vidiu** [comment]
@zenopopovici 
### Patternlab implementation status:
I've implemented the addition of other asset types: images, icons, js. This task needs a review, which will hopefully include what its missing, if that's the case.

File: gulpfile.js/config.js [hunk 1]
```diff
@@ -236,6 +239,76 @@ const modules = {
...
+  // Patternlab-Node
+  patternlab: {
+    paths: {
+      source: {
+        root: paths.modules.patternlab.source.root,
+        patterns: paths.modules.patternlab.source.patterns,
+        data: paths.modules.patternlab.source.root.data,
+        meta: paths.modules.patternlab.source.meta,
+        annotations: paths.modules.patternlab.source.annotations,
+        styleguide: './node_modules/styleguidekit-assets-default/dist/',
```
Comment (zenopopovici) [inline gulpfile.js/config.js]: Create toplevel var for node_modules. (e.g. node.modules)

File: gulpfile.js/config.js [hunk 2]
```diff
@@ -236,6 +239,76 @@ const modules = {
...
+      'views-new': false,
+      'tools-all': false,
+      'tools-docs': false
+    },
+    ishMinimum: '240',
+    ishMaximum: '2600',
+    patternStateCascade: ['inprogress', 'inreview', 'complete'],
+    patternStates: {},
+    patternExportPatternPartials: [],
+    patternExportDirectory: './pattern_exports/',
```
Comment (zenopopovici) [inline gulpfile.js/config.js]: use 'root' toplevel var.

File: gulpfile.js/modules/paths.js [hunk 3]
```diff
@@ -24,6 +24,12 @@ const base = {
   // Wordpress
   wordpress: 'wordpress/',
 
+  // Patternlab
+  patternlab: {
+    source: './src/patternlab/',
```
Comment (zenopopovici) [inline gulpfile.js/modules/paths.js]: use 'src' toplevel var.

File: gulpfile.js/modules/paths.js [hunk 4]
```diff
@@ -24,6 +24,12 @@ const base = {
...
+    public: './www/patternlab/'
```
Comment (zenopopovici) [inline gulpfile.js/modules/paths.js]: use 'root' toplevel var.

File: gulpfile.js/config.js [hunk 5]
```diff
@@ -236,6 +239,76 @@ const modules = {
...
+  patternlab: {
+    paths: {
+      source: {
+        root: paths.modules.patternlab.source.root,
+        patterns: paths.modules.patternlab.source.patterns,
+        data: paths.modules.patternlab.source.root.data,
+        meta: paths.modules.patternlab.source.meta,
+        annotations: paths.modules.patternlab.source.annotations,
+        styleguide: './node_modules/styleguidekit-assets-default/dist/',
+        patternlabFiles: './node_modules/styleguidekit-mustache-default/views/',
```
Comment (zenopopovici) [inline gulpfile.js/config.js]: use 'node.modules' toplevel var.

**Pull request merged**